kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h

```diff
@@ -1327,6 +1371,118 @@ class ResidualBasedBlockBuilderAndSolver
         }
     }
 
+    /**
+     * @brief This method returns the scale norm considering for scaling the diagonal
+     * @param rA The LHS matrix
+     * @return The scale norm
+     */
+    double GetScaleNorm(TSystemMatrixType& rA)
+    {
+        if (mOptions.Is(SCALE_DIAGONAL) ) {
+            if (mScaleFactor > 1.0) {

```

Comment:
why you only allow the "scale_factor" to be > 1.0? it may well be below one...as long as it is not 0 it is ok.

as an aside, why "SCALE_DIAGONAL"? shouldn't it be something of the type "CONSIDER_PRESCRIBED_VALUE" ?

in short i think that this should be something like

~~~
if(m_options.Is(NO_SCALING) ) return 1.0;
else if(mOptions.Is(CONSIDER_PRESCRIBED_DIAGONAL) ) return mScaleFactor;
else if(mOptions.Is(CONSIDER_NORM_DIAGONAL) ) 
                    return GetDiagonalNorm(rA)/static_cast<double>(rA.size1());
else 
                    return GetMaxDiagonal(rA);
~~~